### PR TITLE
<TimeTable/> - new component

### PR DESCRIPTION
**Pull request by AndriiTsarenko**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -250,8 +250,7 @@
   "lint-staged": {
     "linters": {
       "*.{js,scss}": [
-        "yoshi lint --fix",
-        "git add"
+        "yoshi lint --fix"
```
Comment (mykas) [inline package.json]: Is this somehow related?

File: src/DragAndDrop/Draggable/components/Container.js [hunk 2]
```diff
@@ -32,6 +32,12 @@ const target = {
       index: monitor.getItem().index,
     };
   },
+  canDrop(props, monitor) {
```
Comment (mykas) [inline src/DragAndDrop/Draggable/components/Container.js]: This feature news to be added on a separate PR. You are creating TimeTable component on this PR and it shouldn't modify other code.

File: src/SortableList/SortableList.js [hunk 3]
```diff
@@ -8,19 +8,34 @@ import Container from '../DragAndDrop/Draggable/components/Container';
 import times from '../utils/operators/times';
 import withDNDContext from './withDNDContext';
 
+export const INSERTION_RULES = {
```
Comment (mykas) [inline src/SortableList/SortableList.js]: Again we have a feature added on the same PR as TimeTable. Should be a separate pr

File: package.json (hunk 1 above)
Comment (AndriiTsarenko) [reply package.json]: I wonder of this because seem `git add` is not valid operation maybe `git add .`. But I it's not related to my PR and I committed this accidentally.

Comment (AndriiTsarenko) [reply package.json]: I've reverter.

File: src/TimeTable/docs/examples/TimeTableStaticItems.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  ensureDataForDay = (daysData, dayId) => {
+    if (!daysData[dayId]) {
+      daysData[dayId] = [];
+    }
+  };
+
+  handleDrop = (dropInfo) => {
+    const {payload, addedIndex, removedIndex, addedToContainerId, removedFromContainerId} = dropInfo;
```
Comment (mykas) [inline src/TimeTable/docs/examples/TimeTableStaticItems.js]: So this information is too much for our users. If we think about this interface it is not clear for me what it is about. For example what is `payload` in the context of TimeTable.Card ?

File: src/TimeTable/TimeTable.uni.driver.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { baseUniDriverFactory } from 'wix-ui-test-utils/base-driver';
+import { DataAttributes, DataHooks } from './constants';
+
+export const timeTableDriverFactory = base => {
+  const getDataHookSelector = dataHook => `[data-hook="${dataHook}"]`;
+  return {
+    ...baseUniDriverFactory(base),
+    /** returns displayed days count */
+    async daysCount() {
```
Comment (mykas) [inline src/TimeTable/TimeTable.uni.driver.js]: If you look at our other uni driver implementation the convention is not this. Its `daysCount: () => etc`

File: src/TimeTable/TimeTable.js [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+            return (
+              <div
+                key={`${column.title}${column.subtitle}`}
+                data-hook={DataHooks.Header}
+                {...{
+                  [DataAttributes.HeaderActive]: active.toString(),
+                  [DataAttributes.HeaderDisabled]: disabled.toString(),
+                }}
+                className={classNames(styles.header, {
+                  [styles.dayActive]: active,
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: This should be used not with `classNames` library but rather implemented with stylable states. Contact me if you don't know how to do it. Or check this example: https://github.com/wix/wix-style-react/blob/master/src/Text/Text.js#L52

File: src/TimeTable/TimeTable.js [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    );
+  };
+
+  renderInsertButton = containerId => {
+    return (
+      <div
+        onClick={() => this.props.onAddButtonClick(containerId)}
+        className={styles.addItemButton}
+      >
+        <div>
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: Why this additional `<div>` ?

File: src/TimeTable/TimeTable.js [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    groupName: 'time-table',
+  };
+
+  state = {
+    showAddButtonDayIndex: null,
+    dragging: false,
+  };
+
+  static getMonday = date => {
+    date = new Date(date);
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: I noticed that you are mutating values all the time. In our code we are developing with immutability in practise in order to prevent overriding variables. Please change this to be immutable

File: src/TimeTable/TimeTable.js [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+  };
+
+  getRenderData = () => {
+    const { renderOptions, groupName } = this.props;
+
+    if (!renderOptions) {
+      return [];
+    }
+
+    const daysInfo = [];
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: again you are mutating values with this variable. This whole block of code must be done with immutability in mind

File: src/TimeTable/TimeTable.js [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+
+  render() {
+    const { dataHook, groupName } = this.props;
+    const renderData = this.getRenderData();
+    return (
+      <div className={styles.root} data-hook={dataHook}>
+        <div className={styles.headerContainer}>
+          {renderData.map(column => {
+            const {
+              active = false,
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: default values needs to stay in defaultProps - not in the code.

File: src/TimeTable/TimeTable.js [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+    insertPosition: 'any',
+  };
+
+  state = {
+    addButtonColumnIndex: null,
+    dragging: false,
+    sortableListItems: getSortableListItems(this.props.columns),
+  };
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (mykas) [inline src/TimeTable/TimeTable.js]: Isn't this gonna be deprecated?

Comment (simast) [reply src/TimeTable/TimeTable.js]: I thought they just deprecated old methods without `UNSAFE`, but I see they do call the `UNSAFE` methods as legacy now in official documentation as well. I will replace this with `getDerivedStateFromProps`.
